cfme/services/catalogs/catalog_items/__init__.py

```diff
@@ -483,8 +483,33 @@ def create(self, catalog_item_class, *args, **kwargs):
         Returns:
             An instance of catalog_item_class
         """
+        if kwargs:
+            provider = kwargs.get("provider", None)
+            field_entry_point = kwargs.get("field_entry_point", None)
+            if provider and field_entry_point:
+                """kwargs should not contain value of 'provider' and 'field_entry_point'
+                    while instantiating cat_item. Hence released 'provider' and 'field_entry_point'
+                    elements from kwargs.
+                """
+                kwargs.pop("provider")
+                kwargs.pop("field_entry_point")
         cat_item = self.instantiate(catalog_item_class, *args, **kwargs)
-        view = navigate_to(cat_item, 'Add')
+        view = navigate_to(cat_item, "Add")
+        if field_entry_point:
+            prov_type = "Cloud" if provider.category == "cloud" else "Infrastructure"
+            view.basic_info.field_entry_point.fill("")
+            view.basic_info.modal.tree.click_path(
```

Comment:
This is all hardcoded fill values that are specific to a test or a set of assumptions.  They can be kept as a default value (given the updates I describe below)

Overall, this does not seem like the correct way to approach setting this value for a catalog item within its CRUD method.

field_entry_point should be an attribute of the catalog item, passed to its instantiate method as a kwarg.  With it passed directly to create, there's no modeling of the value after the catalog item has been created - we don't have a reference for what the value should be in the UI, we only have what's actually displayed.

With field_entry_point existing as a CloudInfraCatalogItem attribute, it can be included in the `fill_dict` property that's used in this `create` method to fill the form.  field_entry_point is already a widget defined on this form.

If the widget does not fill correctly, as you appear to be doing some custom fill actions here (filling with emptry sting, then using the modal tree) we can evaluate the structure of the fill dict and the catalog item attributes so that they are set and then fill correctly.  We may need to use before_fill for that view.

Reply:
@mshriver 
- As you said, I tried to remove all hard-coded fill values that are specific to  this test.
- Instead of using ```before_fill``` method or including ```field_entry_point``` to ```fill_dict```. I have added [three checks for three path values](https://github.com/ManageIQ/integration_tests/pull/7903/files#diff-3a85f436b2ce6046c949bfc47f9b8b73R488) in create method.
- Let me know, if this is a better way to go with or we must have to either add ```field_entry_point``` to ```fill_dict``` or use ```before_fill``` method.


Thank you for this explanation.